src/transaction/TransactionController.ts

```diff
@@ -470,9 +471,11 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
     const { from } = transactionMeta.transaction;
```

Comment:
It might be safer to wrap this entire section, from line 466 down to 481, in a `try ... finally` block, just in case there is a crash here. Or the `try` block below could be extended to include this section.

For example, if `transactionMeta.transaction` here is undefined this would crash, resulting in this lock never being released.

Reply:
true, I'll add it